### PR TITLE
Fix #149: Add endpoint for all checks by project

### LINKED ISSUE #149: Add endpoint to return result of all checks
We could add an endpoint `/checks/{project}`  that would return the aggregation of all results of a project.

It would use the cached values if any available

```
[{
  "name": "preview-signatures",
  "project": "remotesettings",
  "module": "checks.remotesettings.validate_signatures",
  "description": "Signatures validation of preview collections",
  "documentation": "Signatures should be valid for each collection content.\n\nThe errors are returned for each concerned collection.",
  "url": "/checks/remotesettings/preview-signatures",
  "ttl": 3600,
  "parameters": {},
  "datetime": "2019-10-30T15:12:09.266873",
  "success": true,
  "data": {}
}, {
...
}]
```

### PR DESCRIPTION
**Pull request by leplatrem**
Fix #149

### CODE REVIEW

**Comment by leplatrem** [comment]
It's based on #165. Review by commit shouldn't be too painful :pray:

File: poucave/app.py [hunk 1]
```diff
@@ -19,6 +19,32 @@
...
 
+class Checks:
+    def __init__(self, conf):
+        self.all = []
+        for project, checks in conf["checks"].items():
+            for name, params in checks.items():
+                check = Check(project, name, **params)
+                self.all.append(check)
+
+    def get(self, project=None, name=None):
```
Comment (glasserc) [inline poucave/app.py]: You proposed `lookup` in a video call, and I think that makes more sense (it fails if missing). But I think we should define another method for `get_all`.

File: poucave/app.py [hunk 2]
```diff
@@ -19,6 +19,32 @@
 HTML_DIR = os.path.join(os.path.dirname(os.path.realpath(__file__)), "html")
 
 
+class Checks:
+    def __init__(self, conf):
+        self.all = []
+        for project, checks in conf["checks"].items():
+            for name, params in checks.items():
+                check = Check(project, name, **params)
+                self.all.append(check)
```
Comment (glasserc) [inline poucave/app.py]: Parsing out from `conf` seems a little odd, can we maybe make this a `classmethod` called `from_conf` or something?

File: poucave/app.py [hunk 3]
```diff
@@ -144,6 +152,43 @@ def override_params(self, params: Dict[str, Any]):
...
+    cache = request.app["poucave.cache"]
+
+    try:
+        selected = checks.get(**request.match_info)
+    except ValueError:
+        raise web.HTTPNotFound()
+
+    body = []
+    for check in selected:
+        result = cache.get(check.cache_key)
```
Comment (glasserc) [inline poucave/app.py]: Should we be handling overridden parameters from URLs?

File: poucave/app.py [hunk 4]
```diff
@@ -144,6 +152,43 @@ def override_params(self, params: Dict[str, Any]):
...
+        if result is None:
+            # This check never ran successfully.
+            body.append(check.infos)
+            continue
```
Comment (glasserc) [inline poucave/app.py]: I feel like we should be executing checks as normal here.

File: poucave/app.py (hunk 3 above)
Comment (leplatrem) [reply poucave/app.py]: I don't think so... that would mean the parameters with the same names would have the same value on every check?

**Pull request merged**